domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmAddHandler.java

```diff
@@ -674,18 +675,17 @@ private void addTrustManagerService(OperationContext context, ModelNode ssl, Ser
         final ServiceBuilder<TrustManager[]> serviceBuilder;
         char[] keystorePassword = KeystoreAttributes.KEYSTORE_PASSWORD.resolveModelAttribute(context, ssl).asString()
                 .toCharArray();
-        String provider = KeystoreAttributes.KEYSTORE_PROVIDER.resolveModelAttribute(context, ssl).asString();
-
-        if (JKS.equals(provider) == false) {
-            ProviderTrustManagerService trustManagerService = new ProviderTrustManagerService(provider, keystorePassword);
+        final String provider = KeystoreAttributes.KEYSTORE_PROVIDER.resolveModelAttribute(context, ssl).asString();
 
+        if (!JKS.equalsIgnoreCase(provider)) {
+            final ProviderTrustManagerService trustManagerService = new ProviderTrustManagerService(provider, keystorePassword);
             serviceBuilder = serviceTarget.addService(serviceName, trustManagerService);
         } else {
-            String path = KeystoreAttributes.KEYSTORE_PATH.resolveModelAttribute(context, ssl).asString();
+            final String path = KeystoreAttributes.KEYSTORE_PATH.resolveModelAttribute(context, ssl).asString();
             ModelNode relativeToNode = KeystoreAttributes.KEYSTORE_RELATIVE_TO.resolveModelAttribute(context, ssl);
             String relativeTo = relativeToNode.isDefined() ? relativeToNode.asString() : null;
 
-            FileTrustManagerService trustManagerService = new FileTrustManagerService(provider, path, relativeTo, keystorePassword);
+            final FileTrustManagerService trustManagerService = new FileTrustManagerService(JKS.toLowerCase(Locale.ENGLISH), path, relativeTo, keystorePassword);
```

Comment:
Isnt with this change other then "JKS" truststore forbidden? Wouldn't be provider.toLowerCase better?